airflow/jobs/scheduler_job.py

```diff
@@ -700,22 +728,29 @@ def _process_dags(self, dagbag, dags, tis_out):
                 continue
 
             self.log.info("Processing %s", dag.dag_id)
+            dag_id = dag.dag_id
+            dag_runs_for_dag = list(dag_runs_by_dag_id[dag.dag_id]) if dag_id in dag_runs_by_dag_id else []
```

Comment:
I think `dag_runs_by_dag_id[dag.dag_id]` is already a list so no need for another cast?

Reply:
```suggestion
            dag_runs_for_dag = list(dag_runs_by_dag_id[dag_id]) if dag_id in dag_runs_by_dag_id else []
```

Reply:
Fixed. Thanks.